agent/rpm/pbench-agent.spec.j2

```diff
@@ -8,61 +8,78 @@ URL:            https://github.com/distributed-system-analysis/pbench
 Buildarch:      noarch
 
-# EPEL provides ansible (a curated set of roles with a dependency on ansible-core)
-# on RHEL 8 and RHEL9. Only CentOS-Stream seems to not have an ansible package
-# available.
 
-%if 0%{?centos} >= 8
-Requires:  ansible-core
+# RHEL7, RHEL9 (and CentOS-Stream-9) and the Fedoras provide a python3 package
+# through the standard distro repos.
+# RHEL8 (and CentOS-Stream-8) provide python36, python38 and python39 as modules.
+# We stick with python39 for them for now.
+# N.B. The condition catches both RHEL8 and CentOS-Stream-8)
+%if 0%{?rhel} == 8
+
+Requires:  python39
+# This is used by the shebang processor
+%define __python3 /usr/bin/python3.9
+
 %else
-Requires:  ansible
+
+Requires:  python3
+Requires:  python3-docutils
 %endif
 
 
+# RHEL7 is a special case: it does not conform to the (slightly) more general pattern of the rest
 %if 0%{?rhel} == 7
-Requires:  python3, python3-pip
-%endif
 
-# This condition will be true on a CentOS-Stream-8 system as well
-%if 0%{?rhel} == 8
-Requires:  python36, python3-pip
 # RPMs for modules in requirements.txt
-Requires:  python3-cffi, python3-click, python3-requests
-# RPMs for module dependencies
-Requires:  python3-docutils, python3-psutil
-%endif
+Requires: python3-pip, python3-requests
+#RPMS for module dependencies
+Requires: python3-psutil
+
+%else
+
+{# Jinja2 template for python dependencies - see the targets.json file for the input data #}
+{%for tname, tdata in targets.items() -%}
 
-# This condition will be true on a CentOS-Stream-9 system as well
-%if 0%{?rhel} == 9
-Requires:  python3-pip
+%if {{ tdata.condition }}
 # RPMs for modules in requirements.txt
-Requires:  python3-cffi, python3-requests
-# RPMs for module dependencies
-Requires:  python3-docutils, python3-psutil
+Requires: {{ tname }}-pip, {{ tname }}-cffi, {{ tname }}-requests
+#RPMS for module dependencies
+Requires: {{ tname }}-psutil
 %endif
 
+{% endfor %}
+{# end of template #}
 
+%endif
+
+
+# additional packages that Fedora builds but the RHELs don't
 %if 0%{?fedora} != 0
-Requires:  python3, python3-pip
 # RPMs for modules in requirements.txt
-Requires:  python3-bottle, python3-cffi, python3-click, python3-daemon
-Requires:  python3-jinja2, python3-redis, python3-requests, python3-sh
+Requires:  python3-bottle, python3-click, python3-daemon
+Requires:  python3-jinja2, python3-redis, python3-sh
+%endif
 
 %if 0%{?fedora} >= 36
 Requires: python3-ifaddr
 %endif
 
-# RPMs for module dependencies
-Requires:  python3-psutil
-%endif
-
+# Common requirements
 Requires:  perl, perl-Data-UUID, perl-JSON, perl-JSON-XS
 Requires:  perl-Time-HiRes
 
 Requires:  bc, bzip2, hostname, iproute, iputils, net-tools, numactl
 Requires:  openssh-clients, openssh-server, procps-ng, psmisc, redis
 Requires:  rpmdevtools, rsync, screen, sos, tar, xz
 
+# The condition here should be 7 OR 8.4 OR 8.5 but we don't have a way to deal with minor versions.
+# The resulting RPM will NOT work on RHEL8.[45].
```

Comment:
Sounds like we need to remember to call this out in the release notes. Hopefully it's OK to require 8.6 now? 😒

Reply:
I hope so. No 8.[45], no F36, no 9.1 - the distros are dropping like flies :-)